src/partition/ReceivedBoundingBox.cpp

```diff
@@ -0,0 +1,154 @@
+    }    
+    for (int remoteRank = 0; remoteRank < _remoteParComSize; remoteRank++ )
+    {
+      _globalBB[remoteRank]= initialBB;
+    }
+
+    // master receives global_bb from other master    
+    com::CommunicateBoundingBox(_m2n->getMasterCommunication()).receiveBoundingBoxMap(_globalBB, 0 );
+  }
+}
+
+void ReceivedBoundingBox::computeBoundingBox()
+{
+  // handle coupling mode first (i.e. serial participant)
+  if (not utils::MasterSlave::_slaveMode && not utils::MasterSlave::_masterMode)
+  { 
+    //some code here
+  }
+
+  if (utils::MasterSlave::_slaveMode)
+  {
+    utils::MasterSlave::_communication->broadcast(_remoteParComSize, 0);
+
+    // initializing the _globalBB
+    for (int remoteRank = 0; remoteRank < _remoteParComSize; remoteRank++ )
+    {
+      _globalBB[remoteRank]= _bb;
+    }
+    
+    // receive _globalBB from master
+    com::CommunicateBoundingBox(utils::MasterSlave::_communication).broadcastReceiveBoundingBoxMap(_globalBB);
+    _numberOfRemoteRanks = _globalBB.size();      
+
+    // received bounding boxes are comapred with each rank bb
+    if (_numberOfRemoteRanks>0)
+    {
+      for (auto &other_rank: _globalBB)
+      {
+        if (compareBoundingBox(_bb,other_rank.second)) {
+          _feedback.push_back(other_rank.first);            
+        }
+      }
+    } 
+
+    //send feedback to master
+    utils::MasterSlave::_communication->send(_feedback, 0);     
+  }
+  else if (utils::MasterSlave::_masterMode)
+  { // Master
+    assertion(utils::MasterSlave::_rank==0);
+    assertion(utils::MasterSlave::_size>1);
+    // sending this participant communicator size to other master
+    _m2n->getMasterCommunication()->send(utils::MasterSlave::_size , 0);
+    utils::MasterSlave::_communication->broadcast(_remoteParComSize);
+    com::CommunicateBoundingBox(utils::MasterSlave::_communication).broadcastSendBoundingBoxMap(_globalBB);
+    _numberOfRemoteRanks = _globalBB.size();
+
+    // initializing _feedbackmap
+    for (int rank_slave=1; rank_slave < utils::MasterSlave::_size ; rank_slave++)
+    {
+      _feedbackMap[rank_slave].push_back(-1);        
+    }
+
+    // master produces its own feedback
+    for (auto &other_rank: _globalBB)
+    {
+      if (compareBoundingBox(_bb,other_rank.second))
+      {
+        _feedback.push_back(other_rank.first);
+      }          
+    }
+
+    // master first adds ist feedback to feedbackmap
+    _feedbackMap[0]=_feedback;
+
+    // master receives feedbacks from slaves and put them into the feedbackmap
+    for (int rank_slave=1; rank_slave < utils::MasterSlave::_size ; rank_slave++)
+    {
+      utils::MasterSlave::_communication->receive(_feedback, rank_slave);
+      _feedbackMap[rank_slave]=_feedback;        
+    }
+
+    // master sends feedbackmap to other master
+    com::CommunicateBoundingBox(_m2n->getMasterCommunication()).sendFeedbackMap(_feedbackMap,0);             
+  }
+
+
+  if (utils::MasterSlave::_masterMode)
```

Comment:
Why do we need this block?

Reply:
we need _vertexCounters for final filtering of vertices. Basically, each member of this vector shows the corresponding rank's maximum global vertex id. We filter the vertices first and afterward to decide where each vertice belong, we compare id with the entries of _vertexCounters.